lib/models/mobile_number/mobile_number_error_type.dart

```diff
@@ -0,0 +1 @@
+enum MobileNumberErrorType { unknown, cantCreateMobileNumber, cantConfirmMobileNumber, cantVerifyMobileNumber }
```

Comment:
Can we have these errors on OnboardingPersonalDetailsErrorType enum? 

I would have them in a separate enum only if there was a separate state for the mobile create/verify flow, and that's not the case here